include/villas/nodes/iec61850_goose.hpp

```diff
@@ -0,0 +1,184 @@
+/** Node type: IEC 61850 - GOOSE
+ *
+ * @author Philipp Jungkamp <philipp.jungkamp@rwth-aachen.de>
+ * @copyright 2023, Institute for Automation of Complex Power Systems, EONERC
+ * @license Apache 2.0
+ *********************************************************************************/
+
+#pragma once
+
+#include <string>
+#include <optional>
+#include <cstdint>
+#include <ctime>
+#include <array>
+#include <map>
+#include <villas/node/config.hpp>
+#include <villas/node.hpp>
+#include <villas/pool.hpp>
+#include <villas/queue_signalled.h>
+#include <villas/signal.hpp>
+#include <libiec61850/goose_receiver.h>
+#include <libiec61850/goose_subscriber.h>
+
+namespace villas {
+namespace node {
+namespace iec61850 {
+
+// A GooseValue is a SignalData value with attached Metadata for the MmsType and SignalType
+class GooseValue {
+public:
+	struct MetaSize { int value; };
+
+	// The config file identifier for this type
+	char const * name() const;
+
+	// Corresponding mms type
+	std::optional<MmsType> mmsType() const;
+
+	// Corresponding signal type
+	SignalType signalType() const;
+
+	// Create a GooseValue from an MmsValue
+	static GooseValue fromMmsValue(MmsValue *mms_value);
+
+	// Create a GooseValue from type name and SignalData value
+	static GooseValue fromNameAndValue(char *name, SignalData value);
+
+	// Create a MmsValue from this GooseValue
+	std::optional<MmsValue *> toMmsValue() const;
+
+	// Default/Invalid GooseValue
+	GooseValue();
+
+	union Meta {
+		MetaSize size;
+	} meta;
+
+	SignalData signal_data;
+
+private:
+	struct Descriptor {
+		SignalType signal_type;
+		char const *name;
+		std::optional<MmsType> mms_type;
+		Meta default_meta;
+	};
+
+	inline static std::array const descriptors {
```

Comment:
I already started to build a descriptor table for IEC61850 SV here:

https://github.com/VILLASframework/node/blob/37c9efad92de55f62bd74af97eb755e9b350390e/lib/nodes/iec61850.cpp#L27

I would propose to use / extend the same table for GOOSE ans SV.

Reply:
I'd like to merge the descriptors for Goose and SV. But I think this makes more sense to do while refactoring parts of the SV node as well. Though I should keep the stringified names of the goose and sv nodes in sync.